crates/ruff_linter/src/rules/pycodestyle/rules/logical_lines/extraneous_whitespace.rs

```diff
@@ -134,12 +134,36 @@ pub(crate) fn extraneous_whitespace(
     autofix_before_punctuation: bool,
 ) {
     let mut prev_token = None;
+    let mut fstrings = 0u32;
 
     for token in line.tokens() {
         let kind = token.kind();
+        match kind {
+            TokenKind::FStringStart => {
+                fstrings += 1;
+                continue;
+            }
+            TokenKind::FStringEnd => {
+                fstrings = fstrings.saturating_sub(1);
+                continue;
+            }
+            _ => {}
+        }
         if let Some(symbol) = BracketOrPunctuation::from_kind(kind) {
```

Comment:
Should we just skip all of this when we're inside an f-string?

Reply:
Oh, or actually, we _do_ want to catch some errors within f-strings -- is that right?

Reply:
Removing whitespaces before the punctuations should be fine. It's only when they're removed after when the semantics changes.

So, `f"{foo   :.3f}" and `f"{foo:.3f}"` are equivalent.

Reply:
> Oh, or actually, we _do_ want to catch some errors within f-strings -- is that right?

Yes